javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java

```diff
@@ -387,18 +392,65 @@ protected boolean isModelOrBuiltin(IType type) {
             || ClientModelUtil.isClientModel(type); // client model
     }
 
-    private static String expressionConvertToBinaryData(String name, IType type) {
-        if (type == ClassType.BINARY_DATA) {
-            return name;
-        } else {
-            if (type == ClassType.BASE_64_URL) {
-                return "BinaryData.fromObject(Base64Url.encode(" + name + "))";
-            } else if (type instanceof EnumType) {
-                return "BinaryData.fromObject(" + name + " == null ? null : " + name + "." + ((EnumType) type).getToMethodName() + "())";
-            } else {
-                return "BinaryData.fromObject(" + name + ")";
+    protected enum SupportedMimeType {
+        TEXT,
+        XML,
+        MULTIPART,
+        BINARY,
+        JSON
+    }
+
+    // azure-core SerializerEncoding.SUPPORTED_MIME_TYPES
+    private static final Map<String, SupportedMimeType> SUPPORTED_MIME_TYPES = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
why do we use `TreeMap`?

Reply:
azure-core uses TreeMap and I copied them https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/SerializerEncoding.java#L41

I don't really see the reason why TreeMap is used there as well. It has some difference in ordering when iterating, but their use is also a simple `Map.get`.

Reply:
OK, I get it. It is for ctor with `String.CASE_INSENSITIVE_ORDER`, which is not available in HashMap.

Reply:
Make sure you also do case-insensitive compare with "application/json-merge-patch"